solvers/mathematical_program_result.h

```diff
@@ -218,6 +231,51 @@ class MathematicalProgramResult final {
     return value;
   }
 
+  /**
+   * Gets the dual solution associated with a constraint.
+   *
+   * We interpret the dual variable value as the "shadow price" of the original
+   * problem. Namely if we change the constraint bound by one unit (each unit is
+   * infinitesimally small), the change of the optimal cost is the value of the
+   * dual solution. Mathematically dual_solution = ∂optimal_cost / ∂bound.
+   *
+   * For a linear equality constraint Ax=b where b ∈ ℝⁿ, the dual variables has
+   * n rows, dual_solution(i) is the value of the dual variable for the
+   * constraint A(i,:)*x = b(i).
+   *
+   * For a linear inequality constraint lower <= A*x <= upper where lower and
+   * upper ∈ ℝⁿ, dual_solution also has n rows. dual_solution(i) is the value of
+   * the dual variable for constraint lower(i) <= A(i,:)*x <= upper(i). If
+   * neither side of the constraint is active, then dual_solution(i) is 0. If
+   * the left hand-side lower(i) <= A(i, :)*x is active (meaning lower(i) = A(i,
+   * :)*x at the solution), then dual_solution(i) is positive (because the
+   * objective is to minimize a cost, so the shadow price is positive if the
```

Comment:
Correct, but it took me a second to convince myself. It's actually nothing more than what you already say, but maybe saying explicitly that: since we are minimizing, and the increase of a lower bound can only shrink the feasible set, the cost cannot decrease and the shadow prices must be nonnegative.